pytorch_lightning/core/lightning.py

```diff
@@ -928,20 +934,16 @@ def test_dataloader(self):
 
         .. note:: If you don't need a test dataset and a test_step, you don't need to implement this method.
 
-        .. note:: If you want to change the data during every epoch DON'T use the data_loader decorator.
```

Comment:
why this is removed?

Reply:
I would like the users to rely directly on the `train_dataloader` document. Because of the limitation around dali iterator is somewhat complicated.

Reply:
good idea but with a slight change.

train dataloader support a single dataset.
test/val support multiple datasets. 

so the docs for test/val are the same but not train.